File: DesignPatternsExercises/java/BuilderDesignPattern/Solution/Computer.java
import java.util.ArrayList;

// public class Computer implements __________(21)__________ {
public class Computer implements ComputerPlan {
    private ArrayList<Monitor> monitorList = new ArrayList<>();
    private ArrayList<Ram> ramList = new ArrayList<>();
    private ArrayList<HardDisk> hddList = new ArrayList<>();
    private Mouse mouse;

	@Override
	public void addMonitor(Monitor monitor) {
		monitorList.add(monitor);
	}

	@Override
	public void addRam(Ram ram) {
		ramList.add(ram);
	}

	@Override
	public void addHardDisk(HardDisk hardDisk) {
		hddList.add(hardDisk);
	}

	@Override
	public void addMouse(Mouse mouse) {
		this.mouse = mouse;
	}

	public ArrayList<Monitor> getMonitorList() {
		return monitorList;
	}

	public ArrayList<Ram> getRamList() {
		return ramList;
	}

	public ArrayList<HardDisk> getHddList() {
		return hddList;
	}

	public int getRamCapacity() {
		int capacity = 0;
		for (Ram ram : ramList)
			capacity += ram.getRamCapacity();
		return capacity;
	}

	public int getHddCapacity() {
		int capacity = 0;
		for (HardDisk hardDisk : hddList)
			capacity += hardDisk.getHardDiskCapacity();
		return capacity;
	}

	public Mouse getMouse() {
		return mouse;
	}
}

File: DesignPatternsExercises/ruby/builder_design_pattern/exercise/computer.rb
require_relative 'computer_plan'

class Computer < ComputerPlan
  def __________(1)__________
    super
  end

  def add_monitor(monitor)
    @monitor_list << monitor
  end

  def add_ram(ram)
    @ram_list << ram
  end

  def add_hard_disk(hard_disk)
    @hdd_list << hard_disk
  end

  def add_mouse(mouse)
    @mouse = mouse
  end

  def ram_capacity
    capacity = 0
    @ram_list.each { |ram| capacity += ram.capacity }
    capacity
  end

  def hdd_capacity
    capacity = 0
    @hdd_list.each { |hdd| capacity += hdd.capacity }
    capacity
  end
end